examples/axisartist/demo_curvelinear_grid.py

```diff
@@ -1,3 +1,17 @@
+"""
+Demo of defining curvilinear coordinate using GridHelperCurveLinear and Transformation.
```

Comment:
The docstring looks good overall, but here is a couple of improvements you should make:

* Let's try to make the title shorter: something like "Curvilinear coordinates" would be better.
* Add a sentence at the beginning of the paragraph to introduce it: This examples illustrates how to define curvilinar coordinates using GridHelperCurveLinear and Transformation".